trie/database.go

```diff
@@ -847,3 +852,46 @@ func (db *Database) Size() (common.StorageSize, common.StorageSize) {
 	var metarootRefs = common.StorageSize(len(db.dirties[common.Hash{}].children) * (common.HashLength + 2))
 	return db.dirtiesSize + db.childrenSize + metadataSize - metarootRefs, db.preimagesSize
 }
+
+func (db *Database) saveCache(dir string, threads int) (err error) {
+	if db.cleans == nil {
+		return
+	}
+	defer func(start time.Time) {
+		if err == nil {
+			log.Info("Saved clean cache into the file", "path", dir, "elapsed", common.PrettyDuration(time.Since(start)))
+		} else {
+			log.Info("Failed to save clean cache into file", "error", err)
+		}
+	}(time.Now())
+	err = db.cleans.SaveToFileConcurrent(dir, threads)
+	return err
+}
+
+// SaveCache atomically saves fast cache data to the given dir using a single
+// or multi CPU cores.
+func (db *Database) SaveCache(dir string, concurrent bool) error {
+	threads := 1
+	if concurrent {
+		threads = runtime.GOMAXPROCS(-1)
+	}
+	return db.saveCache(dir, threads)
```

Comment:
You can just use `runtime.GOMAXPROCS(0)` instead of `threads` in `return db.saveCache(dir, threads)`

https://golang.org/pkg/runtime/#GOMAXPROCS

That said, I'm unsure there's a need for non maximal concurrency at all.